streams/src/test/java/org/apache/kafka/streams/processor/internals/TopicPartitionMetadataTest.java

```diff
@@ -0,0 +1,115 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import java.nio.ByteBuffer;
+import java.util.Base64;
+import org.junit.Test;
+
+import static org.apache.kafka.streams.processor.internals.TopicPartitionMetadata.with;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class TopicPartitionMetadataTest {
+
+    @Test
+    public void shouldGetPartitonTimeAndProcessorMeta() {
+        final ProcessorMetadata metadata = ProcessorMetadata.emptyMetadata();
+        final String key = "some_key";
+        final long value = 100L;
+        metadata.addMetadata(key, value);
+
+        final TopicPartitionMetadata topicMeta = with(100L, metadata);
+
+        assertThat(topicMeta.partitionTime(), is(100L));
+        assertThat(topicMeta.processorMetadata(), is(metadata));
+    }
+
+    @Test
+    public void shouldDecodeVersionOne() {
+        final byte[] serialized = ByteBuffer.allocate(Byte.BYTES + Long.BYTES)
+            .put((byte) 1)
+            .putLong(100L)
+            .array();
+        final String serializedString = Base64.getEncoder().encodeToString(serialized);
+
+        final TopicPartitionMetadata topicMeta = TopicPartitionMetadata.decode(serializedString);
+
+        assertThat(topicMeta.partitionTime(), is(100L));
+        assertThat(topicMeta.processorMetadata(), is(ProcessorMetadata.emptyMetadata()));
+    }
+
+    @Test
+    public void shouldEncodeDecodeVersionTwo() {
+        final ProcessorMetadata metadata = ProcessorMetadata.emptyMetadata();
+        final String key = "some_key";
+        final long value = 100L;
+        metadata.addMetadata(key, value);
+
+        final TopicPartitionMetadata expected = with(100L, metadata);
+        final String serializedString = expected.encode();
+        final TopicPartitionMetadata topicMeta = TopicPartitionMetadata.decode(serializedString);
+
+        assertThat(topicMeta, is(expected));
+    }
+
+    @Test
+    public void shouldEncodeDecodeNullMetaVersionTwo() {
+        final TopicPartitionMetadata expected = with(100L, ProcessorMetadata.emptyMetadata());
+        final String serializedString = expected.encode();
+        final TopicPartitionMetadata topicMeta = TopicPartitionMetadata.decode(serializedString);
+
+        assertThat(topicMeta, is(expected));
+    }
+
+    @Test
+    public void shouldDecodeNullVersionTwo() {
+        final TopicPartitionMetadata expected = with(RecordQueue.UNKNOWN, ProcessorMetadata.emptyMetadata());
+        final TopicPartitionMetadata topicMeta = TopicPartitionMetadata.decode(null);
```

Comment:
I actually believe that the consumer contract is to ensure that the metadata `String` is never `null`? I think it's better to not have this test to not mask a potential bug?